endpoints/authentication/src/main.cpp

```diff
@@ -523,7 +556,7 @@ namespace irods::http::handler
 					return _sess_ptr->send(std::move(res));
 				}
 				// OAuth 2.0 Resource Owner Password Credentials Grant
-				else if (const auto alt_method{iter->value().find("iRODS ")}; alt_method != std::string_view::npos) {
+				else if (const auto alt_method{iter->value().find("iRODS ")}; is_oidc_running_as_client() && alt_method != std::string_view::npos) {
```

Comment:
I think this doesn't do what we want it to...

Reply:
Works as expected.

Reply:
ah, very good.